bukkit/src/main/java/hu/montlikadani/tablist/bukkit/tablist/groups/Groups.java

```diff
@@ -245,31 +260,58 @@ private void updatePlayers() {
 
 			if (gp.update()) {
 				sortedPlayers.remove(gp);
-				addToTabListPlayerList(gp);
+				sortedPlayers.add(gp);
 			}
 		}
 
-		int priority = 0;
-		for (GroupPlayer groupPlayer : sortedPlayers) {
-			setPlayerTeam(groupPlayer, priority);
-			priority++;
-		}
+		sortScoreboards();
 	}
 
-	private void addToTabListPlayerList(GroupPlayer tlp) {
-		int pos = 0;
+	/**
+	 * This method is used to sort and update players'
+	 * scoreboards. Includes sorting of AFK players.
+	 * If there is no need to change the places of
+	 * groups, they will only be updated in the custom
+	 * name, prefix, suffix and others.
+	 */
+	public void sortScoreboards() {
+		List<GroupPlayer> playerGroups = sortedPlayers.stream()
+				.sorted(Comparator.comparingInt(GroupPlayer::getPriority))
+				.collect(Collectors.toList());
```

Comment:
Please try to use `for-each` loop instead of stream as this method is called frequently.